topics/topic3/shopping-cart/src/main/java/shoppingcart/ShoppingCart.java

```diff
@@ -0,0 +1,42 @@
+package shoppingcart;
+
+import java.util.ArrayList;
+
+public class ShoppingCart implements ServiceShoppingCart {
+
+	private ArrayList<Product> cartProducts;
+	int totalProducts;
+
+	public ShoppingCart() {
+		this.cartProducts = new ArrayList<Product>();
+		this.totalProducts = 0;
+	}
+
+	@Override
+	public ArrayList<Product> getCartProducts() {
+		return cartProducts;
+	}
+
+	@Override
+	public void addProdcut(Product productToAdd) {
+		cartProducts.add(productToAdd);
```

Comment:
you have to add the product on the first position.
if the product exists into the array.. dont add it, just move it to the first position.